### PR TITLE
Fail Apply/Preview for Required Setters

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent 

This PR is the follow up PR for https://github.com/GoogleContainerTools/kpt/pull/780. The motivation of this PR is to check for the required setters and fail apply/preview in case any of them are not set.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef // indirect
 	github.com/google/uuid v1.1.1
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/cobra v0.0.5
+	github.com/spf13/cobra v1.0.0
```
Comment (mortent) [inline go.mod:10]: This seems like something we certainly should update, but can we do this in a separate PR instead of slipping it in here?

Comment (phanimarupaka) [reply go.mod:10]: It is updated by default when I tried to pull latest kyaml master. Is there any problem due to this ?

Comment (mortent) [reply go.mod:10]: Ah, it should be fine.
